src/plugins/layout.ts

```diff
@@ -86,7 +108,15 @@ export default class Layout implements IPlugin {
   // Add node to the ShadowDom to store initial adjacent node info in a layout and obtain an index
   private discoverNode(node: Node): INodeInfo {
     let shadowNode = this.shadowDom.insertShadowNode(node, getNodeIndex(node.parentNode), getNodeIndex(node.nextSibling));
-    return shadowNode.computeInfo();
+    return this.computeInfo(shadowNode);
+  }
+
+  private computeInfo(shadowNode) {
+    let info = shadowNode.computeInfo();
+    if ((shadowNode.node as Element).tagName === "STYLE" && shadowNode.node.textContent.length === 0) {
+       info.state.cssRules = this.cssRules(shadowNode.node as Element);
```

Comment:
>info.state.cssRules = this.cssRules(shadowNode.node as Element); [](start = 7, length = 64)

This should be somewhere inside already existing stateprovider, so that shadownNode.computeInfo() produces complete state already, including cssRules

Reply:
Agreed and done.

---
In reply to: [243075178](https://github.com/Microsoft/clarity-js/pull/116#discussion_r243075178) [](ancestors = 243075178)